### PR TITLE
Update the reviewer bot to notify based on draft reopen time in addition to event time

### PR DESCRIPTION
**Pull request by ScottSuarez**
closes https://github.com/hashicorp/terraform-provider-google/issues/20142

**Release Note Template for Downstream PRs (will be copied)**

See [Write release notes](https://googlecloudplatform.github.io/magic-modules/contribute/release-notes/) for guidance.

```release-note:none

```

### CODE REVIEW

File: .ci/magician/cmd/scheduled_pr_reminders.go
Comment (ScottSuarez) [reply .ci/magician/cmd/scheduled_pr_reminders.go:359]: We only consider the most recent review from a core reviewer. See the case:switch on line 340.

Comment (melinath) [reply .ci/magician/cmd/scheduled_pr_reminders.go:359]: The behavior I described does _theoretically_ exist - here's a test case that demonstrates it:

```go


		"changes_requested before ready_for_review with approved after": {
			pullRequest: &github.PullRequest{
				User:      &github.User{Login: github.String("author")},
				CreatedAt: &github.Timestamp{time.Date(2024, 1, 1, 0, 0, 0, 0, time.UTC)},
			},
			issueEvents: []*github.IssueEvent{
				&github.IssueEvent{
					Event:             github.String("review_requested"),
					CreatedAt:         &github.Timestamp{time.Date(2024, 1, 1, 0, 0, 0, 0, time.UTC)},
					RequestedReviewer: &github.User{Login: github.String(firstCoreReviewer)},
				},
				&github.IssueEvent{
					Event:     github.String("ready_for_review"),
					CreatedAt: &github.Timestamp{time.Date(2024, 1, 3, 0, 0, 0, 0, time.UTC)}, // Earlier ready
				},
			},
			reviews: []*github.PullRequestReview{
				&github.PullRequestReview{
					User:        &github.User{Login: github.String(firstCoreReviewer)},
					State:       github.String("CHANGES_REQUESTED"),
					SubmittedAt: &github.Timestamp{time.Date(2024, 1, 2, 0, 0, 0, 0, time.UTC)},
				},
				&github.PullRequestReview{
					User:        &github.User{Login: github.String(secondCoreReviewer)},
					State:       github.String("APPROVED"),
					SubmittedAt: &github.Timestamp{time.Date(2024, 1, 4, 0, 0, 0, 0, time.UTC)},
				},
			},
			expectState: waitingForMerge,
			expectSince: time.Date(2024, 1, 4, 0, 0, 0, 0, time.UTC),
		},
```

However, after thinking it over, I think the behavior you have here makes sense... if a change was requested, and then the PR was closed, left for a long time, and then reopened (somehow without a review being requested) we'd want to keep it in "Changes requested" but just reset the clock. (But actually ready_for_review requests a new review, so the only case that can _actually_ happen here is that a review was requested and we reset the clock for when the review should have started.)

**Review by melinath — APPROVED** [review]

File: .ci/magician/cmd/scheduled_pr_reminders.go
Comment (ScottSuarez) [reply .ci/magician/cmd/scheduled_pr_reminders.go:359]: hmm I think there actually is an issue here with dismissed reviews not clearing an earlier changes_requested.

Theoretically the new reviewer should dismiss the old reviewer if their feedback is no-longer applicable and/or they can no longer approve.

Comment (melinath) [reply .ci/magician/cmd/scheduled_pr_reminders.go:364]: Is there an issue with dismissed reviews that was not already present? I don't remember exactly why I implemented this as is; I'm happy to change the behavior around dismissed reviews but I'd rather hold it for a separate PR unless there's a specific impact to dismissed reviews from this work, since it may have additional implications.

**Review by melinath — CHANGES_REQUESTED** [review]

**Review by melinath — APPROVED** [review]

**Pull request merged**
